### PR TITLE
IPFS upload fix

### LINKED ISSUE #948: IPFS Upload script does not upload images.
**Describe the bug**
The [IPFS Upload](https://docs.gaming.chainsafe.io/current/prefab-scripts#ipfs-upload) script uploads metadata, but it does not upload images. It ignores the path and filename and only posts the data.

**Steps**
1. Copy the script from the docs and paste it into a new C# script.
2. Update the key, data, bucket id, path, and filename.
3. Run the script.
4. Check your ChainSafe storage bucket.
5. You will see that the image is blank if you download it, and even if you reference the cid, there is no way to view the image.

**Expected behavior**
Ideally, there are two functions. One function to upload the image, and another function to upload the metadata for the image.

### PR DESCRIPTION
**Pull request by sneakzttv**
IPFS upload fixed
Users can now upload an image or metadata to storage or both at the same time.
Cleaned up and abstracted classes to match our current classes.
Models for metadata match standards but will also work with the SDK, will update with marketplace PR if needed.

image

To Test:
Load up the sample scene & scroll to ipfs.
Enter your bucket & API key from storage https://app.storage.chainsafe.io/ into the IPFS sample object on the left.
The metadata is taken from the object in the sample scene and images are prompted for upload at runtime.
Test image, metadata & image & metadata functions.

Result:
Metadata json and images should be reachable via the link returned from the call in the console as well as being visible in the storage bucket dashboard.

closes #948

### CODE REVIEW

File: Packages/io.chainsafe.web3-unity.lootboxes/Chainlink/Runtime/Libraries/ChainSafe.Gaming.Marketplace.dll.meta
Comment (kantagara) [inline Packages/io.chainsafe.web3-unity.lootboxes/Chainlink/Runtime/Libraries/ChainSafe.Gaming.Marketplace.dll.meta:1]: why is marketplace always getting deleted :(

Comment (sneakzttv) [reply Packages/io.chainsafe.web3-unity.lootboxes/Chainlink/Runtime/Libraries/ChainSafe.Gaming.Marketplace.dll.meta:1]: i have no idea, i cloned a new repo, still doing it

Comment (sneakzttv) [reply Packages/io.chainsafe.web3-unity.lootboxes/Chainlink/Runtime/Libraries/ChainSafe.Gaming.Marketplace.dll.meta:1]: added it back in

File: src/UnitySampleProject/ProjectSettings/EditorBuildSettings.asset
Comment (rob1997) [inline src/UnitySampleProject/ProjectSettings/EditorBuildSettings.asset:1]: Could you revert this change, `git checkout main "src/UnitySampleProject/ProjectSettings/EditorBuildSettings.asset"` and push this file

Comment (sneakzttv) [reply src/UnitySampleProject/ProjectSettings/EditorBuildSettings.asset:1]: done

**Review by oleksandrchainsafe — APPROVED** [review]
Good for now, but should be converted into a Web3 Service too in the next versions.

**Review by kantagara — APPROVED** [review]

**Review by juans-chainsafe — APPROVED** [review]
Great work! I like when things are working at the first try.
We just need to clarify in the docs that the dev needs to use the API secret key from Storage and not the ID

Screenshot 2024-05-13 at 9 42 30 AM

Screenshot 2024-05-13 at 9 42 39 AM

Metadata.json content:
`{"Description":"An NFT description","External_url":"https://google.com.ar","Image":"https://ipfs.chainsafe.io/ipfs/QmbgA3d7vGq4zvfMskDWnXxmBSpezdqu1HHpKVm5ukzXTP","Name":"Name of the NFT","attributes":[{"Display_types":["Stamina"],"Trait_types":["Health"],"Values":["5"]},{"Display_types":["Boost Number"],"Trait_types":["Thunder Power"],"Values":["20"]}]}`
